src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1221 @@
+  }
+
+  size_t sz1 = 0, sectionSz = 0;
+  // Get the size of full debug_ip_layout
+  xclGetDebugIpLayout(handle, nullptr, sz1, &sectionSz);
+  if(sectionSz == 0) {
+    std::cout << "INFO: Failed to find any Debug IP Layout section in the bitstream loaded on device. "
+              << "Ensure that a valid bitstream with debug IPs (AIM, LAPC) is successfully downloaded. \n"
+              << std::endl;
+   return;
+  }
+  // Allocate buffer to retrieve debug_ip_layout information from loaded xclbin
+  map.resize(sectionSz);
+  xclGetDebugIpLayout(handle, map.data(), sectionSz, &sz1);
+}
+
+
+debug_ip_layout*
+DebugIpStatusCollector::getDebugIpLayout()
+{
+  if(0 == map.size()) {
+    std::cout << " INFO: Debug IP Data is not populated." << std::endl;
+    return nullptr;
+  }
+  debug_ip_layout* dbgIpLayout = reinterpret_cast<debug_ip_layout*>(map.data());
+  if(0 == dbgIpLayout->m_count) {
+    std::cout << "INFO: Failed to find any Debug IPs in the bitstream loaded on device." << std::endl;
+    return nullptr;
+  }
+  return dbgIpLayout;
+}
+
+
+void 
+DebugIpStatusCollector::printOverview(std::ostream& _output)
+{
+  debug_ip_layout* dbgIpLayout = getDebugIpLayout();
+  if(nullptr == dbgIpLayout)
+    return;
+
+  _output << "Number of IPs found :: " << dbgIpLayout->m_count << std::endl;
+  for(uint64_t i = 0; i < dbgIpLayout->m_count; i++) {
+    if (dbgIpLayout->m_debug_ip_data[i].m_type > maxDebugIpType) {
+      _output << "Found invalid IP in debug ip layout with type "
+              << dbgIpLayout->m_debug_ip_data[i].m_type << std::endl;
+      return;
+    }
+    ++debugIpNum[dbgIpLayout->m_debug_ip_data[i].m_type];
+  }
+
+  std::stringstream sstr;
+  for(uint32_t i = 0; i < maxDebugIpType; i++) {
+    if(0 == debugIpNum[i]) {
+       continue;
+    }
+    sstr << debugIpNames[i] << "(" << debugIpNum[i] << ")  ";
+  }
+
+  _output << "IPs found [<ipname>(<count>)]: " << sstr.str() << std::endl;
+}
+
+void 
+DebugIpStatusCollector::collect()
+{ 
+  getDebugIpData();
+}
+
+void 
+DebugIpStatusCollector::collect(const std::vector<std::string> & _elementsFilter)
+{
+  if(_elementsFilter.size()) {
+    processElementFilter(_elementsFilter);
+  }
+  collect();
+}
+
+void 
+DebugIpStatusCollector::processElementFilter(const std::vector<std::string> & _elementsFilter)
+{
+  // reset debugIpOpt to all "false" and then process given element filter
+  for(uint32_t i = 0; i < maxDebugIpType; ++i) {
+    debugIpOpt[i] = false;
+  } 
+  for(auto& itr : _elementsFilter) {
+    const char* element = itr.c_str();
+    if(0 == strcmp(element, "aim")) {
+      debugIpOpt[AXI_MM_MONITOR] = true;
+    } else if(0 == strcmp(element, "am")) {
+      debugIpOpt[ACCEL_MONITOR] = true;
+    } else if(0 == strcmp(element, "asm")) {
+      debugIpOpt[AXI_STREAM_MONITOR] = true;
+    } else if(0 == strcmp(element, "lapc")) {
+      debugIpOpt[LAPC] = true;
+    } else if(0 == strcmp(element, "spc")) {
+      debugIpOpt[AXI_STREAM_PROTOCOL_CHECKER] = true;
```

Comment:
why not use std::string::operator== ?

Reply:
One of the operand here is always a const char* (i.e. "aim", "am" etc.). To use std::string::operator==, a string object would need to be created for those "const char*" s. So, it does not give much of an advantage.
Do you see any issue with the current implementation? If there is an issue, then I could definitely use std::string::operator== here.

Reply:
I am wary of out-smarting the standard library.  It's better to use std::string::operator== and let it do the most efficient comparison.  Incidentally the string comparison operator is overloaded to take second argument as a const char*, so under the hood it may do what you are doing, but we don't know for sure and we shouldn't.

Reply:
okay. Done.